Vehicle.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CodeReviewDemo
+{
+    internal class Vehicle : IVehicle
+    {
+        // Manufacturer of the vehicle
+        public string Manufacturer { get; set; }
+
+        // Model of the vehicle
+        public string model { get; set; }
+
+        // Year of manufacture
+        public int year { get; set; }
+
+        // Price of the vehicle
+        public double Price { get; set; }
+
+        // Starts the vehicle's engine
+        public void StartEngine()
+        {
+            Console.WriteLine("Engine started.");
+        }
+
+        // Stops the vehicle's engine
+        public void StopEngine()
+        {
+            Console.WriteLine("Engine stopped.");
```

Comment:
The variable `speed` should be calculated as `distance / time` instead of `time / distance`. This is crucial to ensure the calculation logic is correct.